### DISCUSSION TITLE
send email on failure

### DISCUSSION THREAD

**Discussion by tooptoop4** [argoproj/argo-workflows#7492] [Q&A] [answered]
https://argoproj.github.io/argo-workflows/workflow-notifications/ seems to suggest argo-workflows can send email on failure of workflow but the linked code just looks like an echo command.

is there any smtp configs similar to https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html?highlight=smtp#smtp ?

seems https://github.com/argoproj/notifications-engine/blob/6e7a43362ff167ded5e250a939c7c4098bdafb9d/pkg/services/email.go is missing from workflows project

**Comment by ryhorB-epam** [comment, marked as answer]
you can use a special image with a mail sending program, include the configuration of this program in the secrets
